### ISSUE TITLE
Traccar Service (Win32) stops after 20 Seconds.

### ISSUE THREAD

**Issue by Cashe18** [tananaev/traccar#2265]
I got an error i was not able to repair so I decided to uninstall Traccar, then delete the folder and start from scratch with fresh DB and all. But the traccar service will not stay on.  Port in use error on log. But everything did work fine for 2 weeks.  Log below.
```
2016-08-28 18:07:55  INFO: Operating system name: Windows 2003 version: 5.2 architecture: x86
2016-08-28 18:07:55  INFO: Java runtime name: Java HotSpot(TM) Client VM vendor: Oracle Corporation version: 25.101-b13
2016-08-28 18:07:55  INFO: Memory limit heap: 247mb non-heap: 0mb
2016-08-28 18:07:55  INFO: Character encoding: UTF-8 charset: UTF-8
2016-08-28 18:07:55  INFO: Version: 3.6-SNAPSHOT
2016-08-28 18:08:06  INFO: Starting server...
2016-08-28 18:08:06  WARN: [4E6A380C] error - Address already in use: bind - BindException (... < TrackerServer:138 < ServerManager:73 < Main:33 < ...)
```
Any advise or help would be GREAT!  Thank you

**Comment by tananaev** [comment]
Are you using official release? It looks like some port is already in use on your system, but for some reason it doesn't say which one (maybe you haven't copied the log properly?).

**Comment by Cashe18** [comment]
Windows 32-bit 3.6 version. Long story short. EVERYTHING was working fine for 2 week but some how I lost admin ability from the admin user name. And the rest of the user accounts were NOT admin. In my attempt to access the Database manually to change the password hash, I gave up. So I decided to uninstall, delete folders, restart and Install from scratch since the data wasn't critical.
The fresh install will traccar Service will not stay running. Stops after 20 seconds.  I will log below Server log and wrapper log.
.......... Server log ........
2016-08-28 18:07:55  INFO: Operating system name: Windows 2003 version: 5.2 architecture: x86
2016-08-28 18:07:55  INFO: Java runtime name: Java HotSpot(TM) Client VM vendor: Oracle Corporation version: 25.101-b13
2016-08-28 18:07:55  INFO: Memory limit heap: 247mb non-heap: 0mb
2016-08-28 18:07:55  INFO: Character encoding: UTF-8 charset: UTF-8
2016-08-28 18:07:55  INFO: Version: 3.6-SNAPSHOT
2016-08-28 18:08:06  INFO: Starting server...
2016-08-28 18:08:06  WARN: [4E6A380C] error - Address already in use: bind - BindException (... < TrackerServer:138 < ServerManager:73 < Main:33 < ...)
.......
..............wrapper log.................
STATUS | wrapper  | 2016/08/28 18:07:38 | --> Wrapper Started as Service
STATUS | wrapper  | 2016/08/28 18:07:38 | Java Service Wrapper Community Edition 32-bit 3.5.28
STATUS | wrapper  | 2016/08/28 18:07:38 |   Copyright (C) 1999-2016 Tanuki Software, Ltd. All Rights Reserved.
STATUS | wrapper  | 2016/08/28 18:07:38 |     http://wrapper.tanukisoftware.com
STATUS | wrapper  | 2016/08/28 18:07:38 | 
STATUS | wrapper  | 2016/08/28 18:07:38 | Launching a JVM...
INFO   | jvm 1    | 2016/08/28 18:07:44 | WrapperManager: Initializing...
INFO   | jvm 1    | 2016/08/28 18:07:56 | [WrapperSimpleAppMain] INFO com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Started.
INFO   | jvm 1    | 2016/08/28 18:07:59 | INFO 8/28/16 6:07 PM: liquibase: Clearing database change log checksums
INFO   | jvm 1    | 2016/08/28 18:08:00 | INFO 8/28/16 6:08 PM: liquibase: Successfully acquired change log lock
INFO   | jvm 1    | 2016/08/28 18:08:00 | INFO 8/28/16 6:08 PM: liquibase: Successfully released change log lock
INFO   | jvm 1    | 2016/08/28 18:08:00 | INFO 8/28/16 6:08 PM: liquibase: Successfully acquired change log lock
INFO   | jvm 1    | 2016/08/28 18:08:03 | INFO 8/28/16 6:08 PM: liquibase: Reading from PUBLIC.DATABASECHANGELOG
INFO   | jvm 1    | 2016/08/28 18:08:03 | INFO 8/28/16 6:08 PM: liquibase: Reading from PUBLIC.DATABASECHANGELOG
INFO   | jvm 1    | 2016/08/28 18:08:03 | INFO 8/28/16 6:08 PM: liquibase: Successfully released change log lock
INFO   | jvm 1    | 2016/08/28 18:08:03 | [WrapperSimpleAppMain] INFO org.eclipse.jetty.util.log - Logging initialized @23786ms
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp: 
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: Encountered an error running main:
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: org.jboss.netty.channel.ChannelException: Failed to bind to: 0.0.0.0/0.0.0.0:5084
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.bootstrap.ConnectionlessBootstrap.bind(ConnectionlessBootstrap.java:204)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.traccar.TrackerServer.start(TrackerServer.java:138)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.traccar.ServerManager.start(ServerManager.java:73)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.traccar.Main.main(Main.java:33)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at java.lang.reflect.Method.invoke(Unknown Source)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.tanukisoftware.wrapper.WrapperSimpleApp.run(WrapperSimpleApp.java:325)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at java.lang.Thread.run(Unknown Source)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: Caused by: java.net.BindException: Address already in use: bind
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at sun.nio.ch.Net.bind0(Native Method)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at sun.nio.ch.Net.bind(Unknown Source)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at sun.nio.ch.DatagramChannelImpl.bind(Unknown Source)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at sun.nio.ch.DatagramSocketAdaptor.bind(Unknown Source)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.socket.nio.NioDatagramPipelineSink.bind(NioDatagramPipelineSink.java:129)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.socket.nio.NioDatagramPipelineSink.eventSunk(NioDatagramPipelineSink.java:77)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendDownstream(DefaultChannelPipeline.java:779)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.SimpleChannelHandler.bindRequested(SimpleChannelHandler.java:299)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.SimpleChannelHandler.handleDownstream(SimpleChannelHandler.java:265)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.DefaultChannelPipeline.sendDownstream(DefaultChannelPipeline.java:591)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendDownstream(DefaultChannelPipeline.java:784)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.handler.logging.LoggingHandler.handleDownstream(LoggingHandler.java:338)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.DefaultChannelPipeline.sendDownstream(DefaultChannelPipeline.java:591)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendDownstream(DefaultChannelPipeline.java:784)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.SimpleChannelHandler.bindRequested(SimpleChannelHandler.java:299)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.SimpleChannelHandler.handleDownstream(SimpleChannelHandler.java:265)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.DefaultChannelPipeline.sendDownstream(DefaultChannelPipeline.java:591)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.DefaultChannelPipeline.sendDownstream(DefaultChannelPipeline.java:582)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.Channels.bind(Channels.java:561)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.channel.AbstractChannel.bind(AbstractChannel.java:197)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	at org.jboss.netty.bootstrap.ConnectionlessBootstrap.bind(ConnectionlessBootstrap.java:198)
INFO   | jvm 1    | 2016/08/28 18:08:06 | WrapperSimpleApp Error: 	... 9 more
STATUS | wrapper  | 2016/08/28 18:08:09 | <-- Wrapper Stopped
.......................................

**Comment by tananaev** [comment]
You have some other process using port 5084. You need to stop that process or change port.

**Comment by Cashe18** [comment]
I'm investigating. I'm wondering if the uninstall of the first instance of Traccar left ports open and the reinstall instance see the port locked. Odd

**Comment by tananaev** [comment]
Port can only be blocked by a process, if the process is running. I doubt that your first instance is running if you removed it.

**Comment by Cashe18** [comment]
Wow... UDP port 5084 PLUS over 2500+ ports are taken by Windows Service DNS.exe . So it's a Windows Server issue and not Traccar. Thank you for the Help.

**Comment by tananaev** [comment]
Very weird. Are you sure it's not a virus or something? Why would it need more than 2500 ports?

**Comment by Cashe18** [comment]
OK. FIXED. This seems to be a known memory leak issue caused from a hotfix from the SP2 update. Easiest Fix for those running Windows server 2003 with DNS/AD. Stop the DNS Server Service. Then Start the Traccar service. Once it's traccar service is started/running, restart the DNS Server Service. The DNS server will notice the used ports and it will skip them. I restarted and it all worked fine.
According to Microsoft the DNS server service will randomly take large UDP blocks that it believes are in rare unused ranges. When the occasional DNS-Flush process happends, the UDP pool can change. This error has been fixed in Vista, Win7, Win2008, and above. Also there is a registry entry so DNS Server NEVER uses a port number but rarely required.
